src/main/java/racingcar/domain/CarFactory.java

```diff
@@ -0,0 +1,20 @@
+package racingcar.domain;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CarFactory {
+
+    public static Cars create(List<String> names) {
+        if (names == null || names.isEmpty()) {
+            throw new IllegalArgumentException("자동차 이름은 적어도 하나 이상이어야 합니다.");
+        }
+
+        List<Car> cars = names.stream()
+                .distinct()
+                .map(Car::new)
+                .collect(Collectors.toList());
+
+        return new Cars(cars);
```

Comment:
`cars` 변수의 할당 없이 좀 더 깔끔하게 처리해볼 수 있을 것 같습니다. 🙇

Reply:
```suggestion
        return new Cars(names.stream()
                .distinct()
                .map(Car::new)
                .collect(Collectors.toList()));
```

위 처럼 따로 변수 할당 하지 않고

 생성자 내에서 스트림을 사용하면 되는군요.